benchmarks/cuda/toy_detector_cuda.cpp

```diff
@@ -0,0 +1,198 @@
+                spacepoints_per_event.size(), mr.main);
+            async_copy(vecmem::get_data(spacepoints_per_event),
+                       spacepoints_cuda_buffer);
+
+            traccc::measurement_collection_types::buffer
+                measurements_cuda_buffer(measurements_per_event.size(),
+                                         mr.main);
+            async_copy(vecmem::get_data(measurements_per_event),
+                       measurements_cuda_buffer);
+
+            // Run seeding
+            seeds_cuda_buffer = sa_cuda(spacepoints_cuda_buffer);
+            stream.synchronize();
+
+            // Run track parameter estimation
+            params_cuda_buffer =
+                tp_cuda(spacepoints_cuda_buffer, seeds_cuda_buffer, B);
+            stream.synchronize();
+
+            // Navigation buffer
+            auto navigation_buffer = detray::create_candidates_buffer(
+                det,
+                device_finding.get_config().navigation_buffer_size_scaler *
+                    copy.get_size(seeds_cuda_buffer),
+                mr.main, mr.host);
+
+            // Run CKF track finding
+            track_candidates_cuda_buffer =
+                device_finding(det_view, field, navigation_buffer,
+                               measurements_cuda_buffer, params_cuda_buffer);
+            stream.synchronize();
+
+            // Run track fitting
+            track_states_cuda_buffer =
+                device_fitting(det_view, field, navigation_buffer,
+                               track_candidates_cuda_buffer);
+            stream.synchronize();
+
+            // Create a temporary buffer that will receive the device memory.
+            auto size = track_states_cuda_buffer.headers.size();
+            std::vector<std::size_t> capacities(size, 0);
+            std::transform(track_states_cuda_buffer.items.host_ptr(),
+                           track_states_cuda_buffer.items.host_ptr() + size,
+                           capacities.begin(),
+                           [](const auto& view) { return view.capacity(); });
+
+            traccc::track_state_container_types::buffer
+                track_states_host_buffer{{size, *(mr.host)},
+                                         {capacities, *(mr.host), mr.host}};
+            host_copy.setup(track_states_host_buffer.headers);
+            host_copy.setup(track_states_host_buffer.items);
+
+            // Copy the device container into this temporary host buffer.
+            vecmem::copy::event_type header_event =
+                copy(track_states_cuda_buffer.headers,
+                     track_states_host_buffer.headers,
+                     vecmem::copy::type::device_to_host);
+            vecmem::copy::event_type item_event = copy(
+                track_states_cuda_buffer.items, track_states_host_buffer.items,
+                vecmem::copy::type::device_to_host);
+        }
+    }
+}
+
+BENCHMARK_MAIN();
```

Comment:
Same here. No need for `BENCHMARK_MAIN()`.

Reply:
OK